### ISSUE TITLE
consider using iconv in libtransmission after all

### ISSUE THREAD

**Issue by ckerr** [transmission/transmission#5245]
### What is the issue?

Inspired by discussion at https://github.com/transmission/transmission/pull/5244#issuecomment-1474442137, complete wtih testcase

> [%83h%83%89%83}] %93%c1%96%bd%8cW%92%b7%81E%91%fc%96%ec%90m%83%8a%83^%81[%83%93%83Y (D-ABC 704x396 DivX511).avi

macOS's `[stringEncodingForData]` generates a more-or-less usable string, but `tr_strv_replace_invalid()` just creates a partial string with a lot of `�` characters.

We've used `iconv()` in the past in libtransmission, but without knowing what source charset to use. We should consider using something like  [uchardet](https://github.com/freedesktop/uchardet) to determine what source charset to use.

Note that restoring `iconv()` to libtransmission brings back the issues discussed @ https://github.com/transmission/transmission/pull/4565; for example see https://github.com/transmission/transmission/issues/4547#issuecomment-1374950003.

### Which application of Transmission?

None

**Comment by KeithGrizzell** [comment]
TL;DR: BUMP!

I noticed this issue on my FreeBSD system when I migrated from version 3 to version 4.0.4 this week. (Details below, if anyone's interested.)

It seems you are (or at least back in March were) aware of this issue, and it's the result of switching from using `iconv()`. For what it's worth, as an end user it would be nice to not have breakage like this (see details below) when upgrading. It would also be nice if the behavior of transmission was the same across different systems.

I suppose this calls for an executive decision to be made as to whether to keep the breaking change or roll it back, but in either case it would be helpful to know what to expect going forward. Also, it would be nice to be able to expect the same behavior across multiple operating systems. That was one of the things I really liked about transmission in the past. It always seemed to be lean, mean, and run on any machine. 😀

#### Details:
I migrated from version 3 to version 4.0.4 on FreeBSD. A couple of my seeding torrents were "broken" when (post-migration)  transmission started looking for filenames with `�` in them instead of the characters it used before. Upon inspecting the "broken" torrent files, I confirmed that they have suggested filenames that aren't completely valid UTF-8. For example, one filename has the character `à` (small a with accent grave; see [here](https://www.fileformat.info/info/unicode/char/00e0/index.htm)), which in UTF-8 should be represented as the two bytes (hex)`C3A0`, but instead it was being represented as one byte (hex) `E0`. (Both of those torrents were created by a program that has a bug, the bug possibly being that it generates UTF-16 or UTF-32 and then drops leading zeroes to create UTF-8, which simply doesn't work in all cases. Or maybe it just assumes a typical Latin codepage and doesn't truly bother with unicode. 🤷)

I confirmed that it is an issue on FreeBSD (4.0.4), but it is not an issue on the MacOS version (4.0.4).
I created (using transmission-create and then a hex editor) a test torrent file (TTT.torrent in the attached zip) that picks up the problem when run through transmission-show. The suggested filename in the torrent file `TTT.torrent` is three characters, in hex `E0E0E0`. Those are bad UTF-8, but often represent the "small a with accent grave" character `à`, mentioned above. On MacOS, `transmission-show TTT.torrent` displays the suggested filename as three UTF-8 characters (2 bytes each): `ààà` (or in hex: `C3A0C3A0C3A0`). On FreeBSD, the output is instead showing `���` (or in hex: `EFBFBDEFBFBDEFBFBD`).
[test-torrent-file.zip](https://github.com/transmission/transmission/files/12706256/test-torrent-file.zip)
